flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/print/DescExtendedStyle.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.utils.print;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.utils.EncodingUtils;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * Print desc extended result.
+ *
+ * <p>For example: (printRowKind is false)
+ *
+ * <pre>
+ *  col_name data_type hasNull
+ *  first    STRING    TRUE
+ *  second   INT       TRUE
+ *  third    STRING    TRUE
+ *
+ *  # Detailed Table Information
+ *  Table Statistics:
+ *                    rowCount:    10
+ *                    fileCount:   2
+ *                    rawDataSize: 100
+ *                    totalSize:   10
+ * </pre>
+ *
+ * From this example, we can see that this print style can print different contents in different
+ * parts and calculate the column widths for different parts respectively.
+ */
+@Internal
+public final class DescExtendedStyle implements PrintStyle {
```

Comment:
I would like introduce a combined `PrintStyle`, which contains multiple `PrintStyle`s, each of which can have its own style.